ds3-sdk/src/test/java/com/spectralogic/ds3client/Ds3Client_Test.java

```diff
@@ -919,4 +920,28 @@ public void getTape() throws IOException, SignatureException {
         assertThat(tape.getId(), is(notNullValue()));
         assertThat(tape.getId(), is(UUID.fromString("c7c431df-f95d-4533-b350-ffd7a8a5caac")));
     }
+
+    @Test
+    public void getObjectVerifyFullPayload() throws IOException, SignatureException {
+        final String jobIdString = "a4a586a1-cb80-4441-84e2-48974e982d51";
+        final Map<String, String> queryParams = new HashMap<>();
+        queryParams.put("job", jobIdString);
+        queryParams.put("offset", Long.toString(0));
+
+        final Map<String, String> responseHeaders = new HashMap<>();
+        responseHeaders.put("Content-Length", "4");
+
+        final ByteArraySeekableByteChannel resultChannel = new ByteArraySeekableByteChannel();
+        final String stringResponse = "Response";
+
+        try {
+            MockNetwork
+                    .expecting(HttpVerb.GET, "/bucketName/object", queryParams, null)
+                    .returning(200, stringResponse, responseHeaders)
+                    .asClient()
+                    .getObject(new GetObjectRequest("bucketName", "object", 0, UUID.fromString(jobIdString), resultChannel));
+        } catch (final ContentLengthNotMatchException ex) {
+            assertThat(ex.getMessage(), is("The Content length for object (4) not match the number of byte read (8)"));
```

Comment:
Instead, you can use the following (but make sure its the right type of exception)
@Test(expected = IllegalArgumentException.class)